Relax theme validation to just print warnings instead of raising

vdonato opened the pull request:
Raising exceptions here now seems like a bad user experience since it's
very likely that the user will get annoyed by things exploding when
they're live-editing config files to see changes get applied.

vdonato committed:
Relax theme validation to just print warnings instead of raising

Raising exceptions here now seems like a bad user experience since it's
very likely that the user will get annoyed by things exploding when
they're live-editing config files to see changes get applied.

karriebear commented on lib/streamlit/config.py:
```diff
@@ -1301,18 +1301,23 @@ def func_with_lock():
-        raise RuntimeError('theme.name cannot be "Auto", "Dark", or "Light".')
+
+    if theme_name and theme_name.lower() in theme.RESERVED_THEME_NAMES:
+        LOGGER.warning('theme.name cannot be "Auto", "Dark", or "Light".')
```
I believe `name` is required so by us clearing it, we're not taking any theming into account so I'm kind of leaning towards error instead of warning?

vdonato replied:
`name` is optional since we default to the name `Custom Theme` on the client if it's unset and a theme is defined (I double-checked that this is the case)

karriebear commented on lib/streamlit/config.py:
```diff
@@ -1301,18 +1301,23 @@ def func_with_lock():
+        LOGGER.warning('theme.name cannot be "Auto", "Dark", or "Light".')
+        set_option("theme.name", "")
 
     if (
```
Should we make this an `elif` instead? By clearing out the name, we could run into this warning too when it could just be the first check failing

vdonato replied:
Partially addressed in the other comment (name is optional), but also the tests check that when name is unset and the theme is otherwise complete, we only get one warning.

Let me also add a test that both warns are printed when both name is invalid and the theme is missing required options for completeness.

vdonato committed:
Add more _validate_theme tests

Also fixed some existing ones that no longer tested what we needed them to
when we changed the function to warn instead of raise.
